api/src/db/model/namespace.ts

```diff
@@ -230,72 +235,82 @@ export default class NamespaceModel extends Model {
     }
   }
 
-  async getProjectSetQuotaSize(
+  async getProfileNamespaceQuotaSize(
     profileId: number,
-    clusterId: number
-  ): Promise<ProjectQuotaSize> {
+    namespace: string
+  ): Promise<NamespaceQuotaSize[]> {
     const query = {
       text: `
-        SELECT quota_cpu_size, quota_memory_size, quota_storage_size, quota_snapshot_size FROM cluster_namespace
-          WHERE namespace_id = $1 AND cluster_id = $2;`,
+      SELECT 
+      quota_cpu_size,quota_memory_size,quota_storage_size,quota_snapshot_size 
+      FROM 
+      cluster_namespace 
+      RIGHT JOIN namespace ON cluster_namespace.namespace_id = namespace.id 
+      WHERE profile_id=$1 AND name=$2 AND archived=$3 ;`,
       values: [],
     };
-
     try {
-      const nsResults: ProjectNamespace[] = await this.findNamespacesForProfile(
-        profileId
-      );
-      const clPromises: Promise<ClusterNamespace[]>[] = nsResults.map((nr) =>
-        this.runQuery({ ...query, values: [nr.id, clusterId] })
-      );
-      const clResults: ClusterNamespace[][] = await Promise.all(clPromises);
+      const result = await this.runQuery({
+        ...query,
+        values: [profileId, namespace, false],
+      });
+      return result;
```

Comment:
this code mixes up `OOP` style JS with everything else which has been functional JS. I think theres a good opportunity to clean the could up so that you don't have to use `this`

Reply:
There are 11 classes rn, i can create a ticket and transfer them into functional components in a separate pr

Reply:
https://app.zenhub.com/workspaces/platform-experience-5bb7c5ab4b5806bc2beb9d15/issues/bcgov/platform-services-registry/582